### PR TITLE
Fix issues from KVM list design QA.

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Fixed issues from canonical-web-and-design/MAAS-squad#1955
  - Changed column visibility at different screen sizes
  - Slightly improved mobile styling
  - Built Popover component, and built extended CPUPopover, RamPopover and StoragePopover components from it

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to KVM list and check that the charts show all the down to tablet screen sizes
- Check that the row height remains consistent on all screen sizes
- Check that there is a popover for the RAM and Storage cells

## Fixes

Fixes canonical-web-and-design/MAAS-squad#1955

### Desktop
0 0 0 0_8400_MAAS_r_kvm (16)

### RAM popover
0 0 0 0_8400_MAAS_r_kvm (19)

### Storage popover
0 0 0 0_8400_MAAS_r_kvm (20)

### Storage popover (multiple)
0 0 0 0_8400_MAAS_r_kvm (21)

### CODE REVIEW

**Comment by Caleb-Ellis** [comment]
@lyubomir-popov Let me know about the red asterisk thing if I should override it here or wait until a vanilla update. Also with the popover I tried implementing it with CSS grid but it got a bit much trying to align the padding while also having separate background colors for the top and bottom. I'm not saying it's not possible, or even that difficult, just that the amount of custom scss seemed a bit excessive. So instead I've just reduced the extra space on either side, while still allowing it to not wrap if the numbers enter double digits.

@amylily1011 I've updated the column widths at different screen sizes and slightly improved the mobile styling. But yeah the issue with the disappearing data... best to check with the core team.

**Comment by lyubomir-popov** [comment]
@Caleb-Ellis no need to override the asterisk, I'll check why it is red in vanilla (there might be a reason).
Sounds goos a bout the popover too.

**Comment by amylily1011** [comment]
@Caleb-Ellis I created a bug on launchpad for those already ;-)

**Comment by cristinadresch** [comment]
@lyubomir-popov to create a new card for design improvements for this

**Comment by lyubomir-popov** [comment]
Issue created [here ](https://github.com/canonical-web-and-design/maas-ui/issues/1432)

File: ui/src/app/kvm/views/KVMList/KVMListTable/KVMListTable.test.tsx [hunk 1]
```diff
@@ -12,17 +27,17 @@ const mockStore = configureStore();
 describe("KVMListTable", () => {
   let initialState;
   beforeEach(() => {
-    initialState = {
-      controller: {
+    initialState = rootStateFactory({
+      controller: controllerStateFactory({
         loaded: true,
         loading: false,
         items: [],
```
Comment (huwshimi) [inline ui/src/app/kvm/views/KVMList/KVMListTable/KVMListTable.test.tsx:34]: Feel free to remove any of these model attributes that aren't required for the test (in the future).

File: ui/src/app/kvm/views/KVMList/KVMListTable/RAMColumn/RAMPopover/RAMPopover.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            </div>
+            <div>Unassigned</div>
+          </div>
+          <div className="ram-popover__secondary">
+            <div className="ram-popover__value" data-test="physical">
+              {`${physicalMemory.value} ${physicalMemory.unit}`}
+            </div>
+            <div>Physical RAM</div>
+            <div className="ram-popover__value">
+              &times;&nbsp;&nbsp;
```
[the remark this answers is not in the document]
Comment (Caleb-Ellis) [reply ui/src/app/kvm/views/KVMList/KVMListTable/RAMColumn/RAMPopover/RAMPopover.tsx:60]: Hahaha pretty wild I know!

File: ui/src/app/kvm/views/KVMList/KVMListTable/KVMListTable.test.tsx (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/kvm/views/KVMList/KVMListTable/KVMListTable.test.tsx:34]: Oh yeah I totally meant to do that but got lost in the copy+pastey sauce

**Pull request merged**